hibernate-rx-core/src/main/java/org/hibernate/rx/event/impl/DefaultRxFlushEntityEventListener.java

```diff
@@ -139,53 +152,80 @@ public void onFlushEntity(FlushEntityEvent event) throws HibernateException {
 
 		final boolean mightBeDirty = entry.requiresDirtyCheck( entity );
 
-		final Object[] values = getValues( entity, entry, mightBeDirty, session );
+		return getValues( entity, entry, mightBeDirty, session ).thenAccept( values -> {
+			event.setPropertyValues( values );
 
-		event.setPropertyValues( values );
+			//TODO: avoid this for non-new instances where mightBeDirty==false
+			boolean substitute = wrapCollections( session, persister, types, values );
 
-		//TODO: avoid this for non-new instances where mightBeDirty==false
-		boolean substitute = wrapCollections( session, persister, types, values );
+			if ( isUpdateNecessary( event, mightBeDirty ) ) {
+				substitute = scheduleUpdate( event ) || substitute;
+			}
 
-		if ( isUpdateNecessary( event, mightBeDirty ) ) {
-			substitute = scheduleUpdate( event ) || substitute;
-		}
+			if ( status != Status.DELETED ) {
+				// now update the object .. has to be outside the main if block above (because of collections)
```

Comment:
@DavideD Again I'm surprised to see collection-related changes, just because I thought we had such extremely limited support for collections.

In particular, we *don't* support making changes to the database via modifying a collection, so I'm very surprised to see collection-related changes to the *flush* functionality. What's going on here, exactly? Is it related to cascades or something?